### PR TITLE
Feature/global variable value customization

### PR DESCRIPTION
**Pull request by Maheshkale447**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

File: Ginger/GingerCoreNETUnitTest/GingerCoreNETUnitTest.csproj [hunk 1]
```diff
@@ -114,6 +113,12 @@
     <ProjectReference Include="..\GingerRuntime\GingerRuntime.csproj" />
   </ItemGroup>
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNETUnitTest/GingerCoreNETUnitTest.csproj:114]: Unit tests can be added to cover it, we have a lot of JSON unit tests which Global Variables section can be added and be validate in some of them

**Review by MeniKadosh1 — DISMISSED** [review]

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs [hunk 2]
```diff
@@ -249,6 +257,44 @@ private void LoadSealightsDetailsFromJSON(GingerExecConfig gingerExecConfig)
...
+                            case "String":
+                            case "Number":
+                            case "Selection List":
+                            case "DateTime":
+                                if (!originalVar.SetValue(customizedVar.VariableCustomizedValue))
+                                {
+                                    throw new InvalidOperationException($"Invalid value/format '{customizedVar.VariableCustomizedValue}' provided for variable '{originalVar.Name}'");
+                                }
+                                break;
+                            default: throw new InvalidOperationException($"Value customization is not allowed for variable '{originalVar.Name}'. Only String, Number,Selection List and DateTime variable value customization is allowed");
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs:283]: Change message as "Value customisation is not supported for variable type {Type of variable}, variable name {Name}

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs [hunk 3]
```diff
@@ -249,6 +257,44 @@ private void LoadSealightsDetailsFromJSON(GingerExecConfig gingerExecConfig)
...
+                {
+                    VariableBase originalVar = DynamicExecutionManager.FindItemByIDAndName<VariableBase>(
+                                   new Tuple<string, Guid?>(nameof(VariableBase.Guid), customizedVar.VariableID),
+                                   new Tuple<string, string>(nameof(VariableBase.Name), customizedVar.VariableName),
+                                   WorkSpace.Instance.Solution.Variables);
+                    if (originalVar != null)
+                    {
+                        switch (originalVar.VariableType)
+                        {
+                            case "String":
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs:274]: why hard coded strings ? can we use enum or typeof(VariableString)

**Review by JinendraGhodke — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs (hunk 3 above)
Comment (Maheshkale447) [reply Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs:274]: We have used Generic function to find variable using either by Id or name, it by default case object to VariableBase so its not possible get actual type of object, VariableType property is already defined in all variable classes to identify the type of object so using it in switch as switch does not support typeof(object)

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs (hunk 2 above)
Comment (Maheshkale447) [reply Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs:283]: done

File: Ginger/GingerCoreNETUnitTest/GingerCoreNETUnitTest.csproj (hunk 1 above)
Comment (Maheshkale447) [reply Ginger/GingerCoreNETUnitTest/GingerCoreNETUnitTest.csproj:114]: sure will add it in different PR

**Review by JinendraGhodke — APPROVED** [review]

**Pull request merged**